pyro/infer/mcmc/hmc.py

```diff
@@ -156,9 +156,9 @@ def _compute_trace_log_prob(self, model_trace):
     def _kinetic_energy(self, r):
         r_flat = torch.cat([r[site_name].reshape(-1) for site_name in sorted(r)])
         if self.full_mass:
-            return 0.5 * r_flat.dot(self._inverse_mass_matrix.matmul(r_flat))
+            return 0.5 * (r_flat * (self._inverse_mass_matrix.matmul(r_flat))).sum()
```

Comment:
LGTM, if it is possible, it is better to add a TODO comment here for easy to remember (because this pattern is hard to grep).